src/initio/initio.c

```diff
@@ -787,7 +787,6 @@ mraa_io_init(const char* strdesc, mraa_io_descriptor** desc)
                     strncat(new_desc->leftover_str, str_descs[i], strlen(str_descs[i]));
                 }
 
-                strncat(new_desc->leftover_str, str_descs[i], strlen(str_descs[i]));
```

Comment:
How is this related to "Add safety checks inside MraaIo destructor"? Although it indeed looks strange given we do the same right above in the `else` branch, I suggest splitting this out to a separate commit to avoid polluting git history by commits where unrelated stuff is crammed together.